frontend/src/lib/lemon-ui/Tooltip/Tooltip.tsx

```diff
@@ -1,61 +1,142 @@
+    FloatingPortal,
+    offset as offsetFunc,
+    Placement,
+    shift,
+    useDismiss,
+    useFloating,
+    useFocus,
+    useHover,
+    useInteractions,
+    useMergeRefs,
+    useRole,
+    useTransitionStyles,
+} from '@floating-ui/react'
+import clsx from 'clsx'
+import React, { useRef, useState } from 'react'
 
-export type TooltipProps = AntdTooltipProps & {
+export interface TooltipProps {
+    title: string | React.ReactNode | (() => string)
+    children: JSX.Element
     delayMs?: number
+    offset?: number
+    arrowOffset?: number
+    placement?: Placement
+    className?: string
+    visible?: boolean
 }
 
-/** Extension of Ant Design's Tooltip that enables a delay.
- *
- * Caveat: doesn't work with disabled elements due to lack of workaround that Ant Design uses.
- * See https://github.com/ant-design/ant-design/blob/master/components/tooltip/index.tsx#L82-L130.
- */
-// CAUTION: Any changes here will affect tooltips across the entire app.
-export function Tooltip({ children, visible, delayMs = DEFAULT_DELAY_MS, ...props }: TooltipProps): JSX.Element {
-    const [localVisible, setVisible] = useState(false)
-    const [debouncedLocalVisible] = useDebounce(visible ?? localVisible, delayMs)
-
-    const floatingContainer = useFloatingContainerContext()?.current
-
-    if (!('mouseEnterDelay' in props)) {
-        // If not preserving default behavior and mouseEnterDelay is not already provided, we use a custom default here
-        props.mouseEnterDelay = delayMs
-    }
-
-    // If child is not a valid element (string or string + ReactNode, Fragment), antd wraps children in a span.
-    // See https://github.com/ant-design/ant-design/blob/master/components/tooltip/index.tsx#L226
+export function Tooltip({
+    children,
+    title,
+    className = '',
+    placement = 'top',
+    offset = 8,
+    arrowOffset,
+    delayMs = 500,
+    visible: controlledOpen,
+}: TooltipProps): JSX.Element {
+    const [uncontrolledOpen, setUncontrolledOpen] = useState(false)
+    const caretRef = useRef(null)
+
+    const open = controlledOpen ?? uncontrolledOpen
+
+    const { context, refs } = useFloating({
+        placement,
+        open,
+        onOpenChange: setUncontrolledOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offsetFunc(offset),
+            flip({ fallbackAxisSideDirection: 'start' }),
+            shift(),
+            arrow({ element: caretRef }),
+        ],
+    })
+
+    const hover = useHover(context, {
+        move: false,
+        delay: {
+            open: delayMs,
+            close: 0,
+        },
+    })
+    const focus = useFocus(context)
+    const dismiss = useDismiss(context)
+    const role = useRole(context, { role: 'tooltip' })
+
+    const { getFloatingProps, getReferenceProps } = useInteractions([hover, focus, dismiss, role])
+
+    const { styles: transitionStyles } = useTransitionStyles(context, {
+        duration: {
+            open: 200,
```

Comment:
100 could feel a bit more responsive, but I don't feel strongly about this

Reply:
I think I tried copying whatever antd did which might be where 200 came from. The 100 went by so quick I felt like I missed the subtle fade in... so split the difference and went with 150 :) Feel free to change because you have a great eye for the details like this